src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -69,6 +69,7 @@ private static int GetIndexOfFirstNonAsciiByteInLane_AdvSimd(Vector128<byte> val
         /// </summary>
         /// <param name="value"></param>
         /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Does the JIT do any better by default if it's instead implemented as:
```C#
return BitConverter.IsLittleEndian ?
    (value & 0xFF80u) == 0 :
    (value & 0xFF800000u) == 0;
```

Reply:
Sadly, no:

![image](https://user-images.githubusercontent.com/523221/229854031-dd4f62f5-a965-4a2e-8274-4b433dd2d4d1.png)

Reply:
Elsewhere in corelib we'll use:
```C#
#if BIGENDIAN
```
Should we do the same thing here instead and not rely on the linker at all?

Reply:
Probably? I think we'll fix that in JIT's inliner eventually so up to you whatever looks better in code